openedx/openedx-platform
# [BB-5968][BD-13] feat: deprecate filestore

## Description

This pull request:

- Removes `filestore` property from the `ModuleSystem` in favor of `runtime.resources_fs` property.

    In the original code, `filestore` is equal to [DescriptorSystem.runtime.resources_fs](https://github.com/openedx/edx-platform/blob/bd43306da43c9b92e35a1f9622bc3c9e9e86fedd/lms/djangoapps/courseware/module_render.py#L772). It's safe to replace it with [ModuleSystem.runtime.resources_fs](https://github.com/openedx/edx-platform/pull/30131/files#diff-77935378e2968d43ac1e33d52e46de7e1af3e9cb9fec96f99bd86e5fff0ffb0cR832), because both runtimes are combined using the `CachingDescriptorSystem` which [provides](https://github.com/openedx/edx-platform/blob/af86e1612c72466eaad7bd8a9125a669faa74d29/common/lib/xmodule/xmodule/modulestore/split_mongo/caching_descriptor_system.py#L63) `resources_fs` property, and therefore in fact the same file storage.
- Renames `filestore` argument to `resources_fs` in the `LoncapaSystem`'s constructor.
- Adds `filestore` deprecate property to the `ModuleSystemShim` and [the same property](https://github.com/openedx/edx-platform/blob/d3f6ed09d8cae2f6430cb4b33de5b981c6cd8066/openedx/core/djangoapps/xblock/runtime/shims.py#L142-L148) of the `RuntimeShim`.

This change should not affect any user of edx-platform.

## Supporting information

- [BB-5968](https://tasks.opencraft.com/browse/BB-5968) - internal OpenCraft ticket.

## Testing instructions

1. Login: https://bb5968.opencraft.hosting (staff@example.com;edx).
2. Open the demo course [outline](https://bb5968.opencraft.hosting/courses/course-v1:edX+DemoX+Demo_Course/course/), try opening different units. You shouldn't see any errors.
3. Go to the Studio, ensure that [units previews](https://studio.bb5968.opencraft.hosting/container/block-v1:edX+DemoX+Demo_Course+type@vertical+block@vertical_0270f6de40fc) are working.
4. Open [the preview](https://preview.bb5968.opencraft.hosting/courses/course-v1:edX+DemoX+Demo_Course/courseware/d8a6192ade314473a78242dfeedfbf5b/edx_introduction/1?activate_block_id=block-v1%3AedX%2BDemoX%2BDemo_Course%2Btype%40vertical%2Bblock%40vertical_0270f6de40fc) view. You shouldn't see errors there as well.
5. Open example [CAPA unit](https://studio.bb5968.opencraft.hosting/container/block-v1:edX+DemoX+Demo_Course+type@vertical+block@11bb317994fb4476abf744792a2170f3?action=new). Check that preview and live versions are working fine.
